Libraries/Dijkstra/Dijkstra.cpp

```diff
@@ -38,50 +38,34 @@ void Dijkstra::updateMatrix(const Map tiles_map) {
     // Then checking and updating weights in the matrix_.
     if (tiles_map.getTile(current_coord_->x, current_coord_->y).ableToGoNorth()
     && !tiles_map.getTile(current_coord_->x - 1, current_coord_->y).isBlack()) {
-      if (tiles_map.getTile(current_coord_->x - 1, current_coord_->y).isVisited()
-      && !matrix_[current_coord_->x - 1][current_coord_->y].checked) {
-        open_vector_.pushAsLast(&matrix_[current_coord_->x - 1][current_coord_->y].current);
-        matrix_[current_coord_->x - 1][current_coord_->y].checked = true;
-      }
       if (matrix_[current_coord_->x - 1][current_coord_->y].weight > matrix_[current_coord_->x][current_coord_->y].weight + tiles_map.getTile(current_coord_->x - 1, current_coord_->y).getWeight()) {
         matrix_[current_coord_->x - 1][current_coord_->y].weight = matrix_[current_coord_->x][current_coord_->y].weight + tiles_map.getTile(current_coord_->x - 1, current_coord_->y).getWeight();
         matrix_[current_coord_->x - 1][current_coord_->y].prev = matrix_[current_coord_->x][current_coord_->y].current;
+        open_vector_.pushAsLast(&matrix_[current_coord_->x - 1][current_coord_->y].current);
```

Comment:
I did it because before I was just updating the adjacent tiles of the ones in the non_visited_vector. But then I realized there were some issues in some specific situations because some tiles were not being updated. I removed the isVisited and not checked conditions because I want to update each tile and its adjacent ones every time its weight changes. Now it does not matter if it has been checked once, it is going to be checked the necessary times.

Reply:
I see, good one.